lib/usdUfe/utils/Utils.cpp

```diff
@@ -31,24 +31,28 @@ std::string sanitizeName(const std::string& name)
 
 std::string prettifyName(const std::string& name)
 {
-    std::string prettyName(1, std::toupper(name[0]));
-    size_t      nbChars = name.size();
-    bool        capitalizeNext = false;
-    for (size_t i = 1; i < nbChars; ++i) {
+    std::string prettyName;
+    prettyName.reserve(name.size());
```

Comment:
If the name is something like "NiceName" the string size will most certainly increase due to the added space in line 43.
Perhaps we could make this a name.size() + 5 to cover us for even a string ThatHasFiveCapitalizedWords.

Not a huge deal though, since there will only be a couple of reallocations at the end.

Reply:
Yeah, I could over-reserve a bit. The old code did not reserve at all, bu yeah, adding a few extra to acount for typical numbers of spaces is easy.

Reply:
Just dropping in a dumb question. Why do we have to reserve the size first? Couldn't we just declare an empty string and appending to it?

Reply:
@samuelliu-adsk since we're adding characters to prettyName over an iteration, C++ doesn't know ahead of time how many we need so it keeps allocating for every new character. We reserve an amount more than needed to do a single allocation for the entire operation. 

In short, allocating one large memory is more efficient than allocation many small memories.